components/validate-email-form.js

```diff
@@ -0,0 +1,20 @@
+// write any logic for your form in one file that can be used on both the client and the server
+
+export default function(email) {
+  // IRL you probably want a better check!
+  if (email.indexOf('@') > -1) {
+    return {
+      valid: true,
+      email,
+      errors: []
+    }
+  } else {
+    return {
+      valid: false,
+      email,
+      errors: [
+        `Email address ${email} is not valid`
+      ]
+    }
+  }
+}
```

Comment:
This is the magic bit - all of your form's logic should be in a file that can be webpacked on the client or used on the server.